core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -2523,6 +2613,7 @@ public BsonValue apply(@NonNull final BsonDocument bsonDocument) {
           undoCollection.deleteOne(getDocumentIdFilter(documentId));
           eventsToEmit.add(event);
         }
+        triggerListeningToNamespace(namespace);
```

Comment:
[q] Do we need to trigger here as a part of this change?

Reply:
Yes, `desync` used to trigger every time it was called. It no longer does that, so we need to do it manually after desyncing has been done. There is a cleaner way to do this by tracking what operations have been done and if the streams need to be restarted.